### DISCUSSION TITLE
/api/v2/search

### DISCUSSION THREAD

**Discussion by SidoKnee** [mastodon/mastodon#23456] [Q&A] [answered]
Hello,
New here.
Trying to do a simple JavaScript code to get/fetch request. I have made an APP & got a Key. But I cannot find any Examples of the Code needed to execute the Get request , All I get back is, 400 or 404 !? With sometimes {error: 'param is missing or the value is empty: q\nDid you …h\n action\n               controller'}.
That is a Call to mastodon.online .
Can anyone Help, Please ? Thank-You :~)
Sid

**Comment by SidoKnee** [comment]
Oh I should Add,
I have Sent, the following code;
headers: {
                 'Authorization': 'Bearer ' + token,
                 'Content-Type': 'application/json',
}
Regards
Sid

**Comment by SidoKnee** [comment]
Oh & here is my latest try of Code --->
"headers": {
                 "Authorization": "Bearer " + token,
                 "Content-Type": "application/json",
                     },
             "param":{
                 "q":"hashtags",
                 "value": "cats",
                 "limit": "5",
                 "resolve": true,
             },
             "method": "get",

Any Help would be Gratefully Appreciated
Regards
Sid

**Comment by andypiper** [comment, marked as answer]
Are you still having issues?

It looks like you're trying to pass a JSON body to the query endpoint, but you need to do a GET with parameters. Here's a simple curl example:

```
curl --header "Authorization: Bearer $TOKEN" https://$INSTANCE/api/v2/search\?q\=cats\&type\=hashtags\&limit\=5\&resolve\=true
```
